src/Database/Adapter/MariaDB.php

```
@@ -867,15 +868,15 @@ public function deleteDocument(string $collection, string $id): bool
         $stmtPermissions->bindValue(':_uid', $id);
 
         try {
-            $stmt->execute() || throw new Exception('Failed to delete document');
-            $stmtPermissions->execute() || throw new Exception('Failed to clean permissions');
+            $stmt->execute() || throw new DatabaseException('Failed to delete document');
+            $stmtPermissions->execute() || throw new DatabaseException('Failed to clean permissions');
         } catch (\Throwable $th) {
             $this->getPDO()->rollBack();
-            throw new Exception($th->getMessage());
+            throw new DatabaseException($th->getMessage());
```

I know it was like this but..
should we Throw the caught Exception?

---

This is intentional so that we can catch our specific exception type on the Appwrite side 👍